ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -795,17 +795,28 @@ abstract class EventStorageBackendTemplate(
   }
 
   override def maxEventSequentialIdOfAnObservableEvent(
-      offset: Offset
-  )(connection: Connection): Option[Long] = {
+      offsetInclusive: Offset
+  )(connection: Connection): Long = {
+    val ledgerEnd = ledgerEndCache()
     import com.daml.platform.store.backend.Conversions.OffsetToStatement
     SQL"""
-         SELECT
-            event_sequential_id_last
-         FROM
-            participant_transaction_meta
-         WHERE
-            event_offset = (SELECT MAX(event_offset) FROM participant_transaction_meta WHERE event_offset <= $offset)
-       """.as(get[Long](1).singleOpt)(connection)
+     SELECT
```

Comment:
The method name doesn't match the sql query. I'd extract this query to a new method

Reply:
Why do you think it does not match?

Reply:
well I think the previous is accurate, this is not: it is not the smallest, but the smallest minus one (that is why the biggest is accurate)

Reply:
I was considering only the sql for this method name. The `- 1` can be that outside